lib/seed/runner.rb

```diff
@@ -157,25 +158,43 @@ def create_progress_bar
       )
     end
 
-    def create_appts(patient_info, facility:, user_ids:)
-      attrs = patient_info.each_with_object([]) { |(patient_id, recorded_at), attrs|
-        number_appointments = config.rand_or_max(0..1) # some patients dont get appointments
-        next if number_appointments == 0
-        scheduled_date = Faker::Time.between(from: Time.current, to: 45.days.from_now)
-        created_at = Faker::Time.between(from: 4.months.ago, to: 1.day.ago)
+    def create_appointments_and_call_results(patient_info, facility:, user_ids:)
+      appointment_attributes = patient_info.map do |(patient_id, _recorded_at)|
+        number_of_appointments = config.rand_or_max(0..1) # some patients dont get appointments
+        next if number_of_appointments.zero?
+
+        device_created_at = Faker::Time.between(from: 6.months.ago, to: 1.day.ago)
+        created_at = Faker::Time.between(from: device_created_at, to: 1.day.ago)
+        scheduled_date = Faker::Time.between(from: device_created_at, to: device_created_at.advance(days: 45))
         user_id = user_ids.sample
-        hsh = {
+        FactoryBot.attributes_for(:appointment,
           creation_facility_id: facility.id,
           facility_id: facility.id,
           patient_id: patient_id,
           scheduled_date: scheduled_date,
+          device_created_at: device_created_at,
           created_at: created_at,
           updated_at: created_at,
-          user_id: user_id
-        }
-        attrs << FactoryBot.attributes_for(:appointment, hsh)
-      }
-      Appointment.import(attrs)
+          user_id: user_id)
+      end.compact
+
+      call_result_attributes = appointment_attributes.map do |appointment|
+        number_of_calls = config.rand_or_max(0..1) # you can get more than 1 call if you are overdue
```

Comment:
Should this be `rand_or_max(0..3)` going by the comment?

Reply:
Technically it would make sense. But looking at the data 50% call rate for appointments is satisfactory. I'll remove the comment to avoid confusion.

Reply:
Make sense.